sdk/dotnet/Pulumi.Automation/LocalWorkspace.cs

```diff
@@ -335,6 +353,39 @@ public static Task<WorkspaceStack> CreateOrSelectStackAsync(LocalProgramArgs arg
 
         private static readonly string[] SettingsExtensions = new string[] { ".yaml", ".yml", ".json" };
 
+        private async Task GetPulumiVersionAsync(CancellationToken cancellationToken = default)
+        {
+            var result = await this.RunCommandAsync(new[] { "version" }, cancellationToken).ConfigureAwait(false);
+            var versionString = result.StandardOutput.Trim();
+            if (versionString[0] == 'v') {
+                versionString = versionString.TrimStart('v');
```

Comment:
`TrimStart` is safe to call even if the string doesn't start with this character, so you don't need the `if` above.